packages/wonder-blocks-testing/src/fixtures/fixtures.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+import * as React from "react";
+import {getConfiguration} from "./setup.js";
+import {combineOptions} from "./combine-options.js";
+
+import type {FixturesOptions, GetPropsOptions} from "./types.js";
+
+type FixtureProps<TProps: {...}> =
+    | $ReadOnly<TProps>
+    | ((options: $ReadOnly<GetPropsOptions>) => $ReadOnly<TProps>);
+
+/**
+ * Describe a group of fixtures for a given component.
+ *
+ * Only one `fixtures` call should be used per fixture file as it returns
+ * the exports for that file.
+ *
+ * @param {FixtureOptions<TProps>} options Options describing the
+ * fixture group.
+ * @param {FixtureFn<TProps> => void} fn A function that provides a `fixture`
+ * function for defining fixtures.
+ * @returns {Exports} The object to be exported as `module.exports`.
+ *
+ * TODO(somewhatabstract): Determine a way around this requirement so we
+ * can support named exports and default exports via the adapters in a
+ * deterministic way. Currently this is imposed on us because of how
+ * storybook, the popular framework, uses both default and named exports for
+ * its interface.
+ */
+export const fixtures = <TProps: {...}>(
+    options: $ReadOnly<FixturesOptions<TProps>>,
+    fn: (
+        fixture: (
+            description: string,
+            props: FixtureProps<TProps>,
+            wrapper?: React.ComponentType<TProps>,
+        ) => void,
+    ) => void,
+): ?$ReadOnly<mixed> => {
+    const {adapter, defaultAdapterOptions} = getConfiguration();
+    const {
+        title,
+        component,
+        description: groupDescription,
+        defaultWrapper,
+        additionalAdapterOptions,
+    } = options;
+
+    // 1. Create a new adapter group.
+    const group = adapter.declareGroup<TProps>({
+        title: title || component.displayName || component.name || "Component",
+        description: groupDescription,
+    });
```

Comment:
I'm not sure what Storybook does with a test title that's the same between different files. I think we might be take more precautions to make this unique.

There's a secondary issue with the organization of stories into groups. Titles with / in them will turn into folders in the Storybook UI. Maybe we could check if there's no / in the title and if there isn't we could put all of those stories into an "Uncategorized" folder by prepending the title with "Uncategorized/". This will keep the number of top-level folders to a minimum.

Reply:
That would be something to do at the top group level rather than here.

Reply:
I didn't realize the groups could be nested.  That's a neat feature.

It could be an optional behaviour.  Maybe I'm being paranoid.  Maybe telling people to organize their stories into folders will be enough and if enough people forget, the result will be so annoying that people will self correct.

Reply:
To be clear, the nesting is a storybook feature based on the slashes in the name. There's no procedural nesting of the groups in the framework itself (at least not yet).